spark_parser/spark.py

```diff
@@ -494,11 +500,16 @@ def makeSet(self, token, sets, i):
 
             if parent == i:
                 continue
-
+            if self.debug['reduce']: #STRANIKS_SCAN 01/03/2016
+                indent2 = len(self.states[state].complete)
             for rule in self.states[state].complete:
                 lhs, rhs = rule
                 if self.debug['reduce']:
-                    print("%s ::= %s" % (lhs, ' '.join(rhs)))
+                    text1 = "%d %d%s%s%s ::= %s" % (indent1, indent2, '   '*indent1, ' '*indent2, lhs, ' '.join(rhs))
```

Comment:
Ok. Thanks for the help. Still it would be great to see examples of what this code does, how it improves things. Since you are in the best position to do that, maybe you could create in your fork, some worked out examples in a wiki. Or create a gist. Thanks.